dartagnan/src/main/java/com/dat3m/dartagnan/program/processing/ProcessingManager.java

```diff
@@ -143,6 +143,15 @@ private ProcessingManager(Configuration config) throws InvalidConfigurationExcep
                 ),
                 RemoveUnusedMemory.newInstance(),
                 MemoryAllocation.fromConfig(config),
+                detectMixedSizeAccesses ? Tearing.fromConfig(config) : null,
+                detectMixedSizeAccesses ? ProgramProcessor.fromFunctionProcessor(
+                        FunctionProcessor.chain(
+                                performAssignmentInlining ? AssignmentInlining.newInstance() : null,
+                                sccp,
+                                dce,
+                                removeDeadJumps
+                        ), Target.THREADS, true
+                ) : null,
```

Comment:
Why do we run all these passes again?

Reply:
Because Tearing opens up new possibilities to optimize you had mixed-size accesses on local variables. At least `Mem2Reg` and `SCCP` should run again after Tearing. If the set of optimization is identical to what we already had, we can refactor this into a single local variable and reuse it twice.

Reply:
I think `MemToReg` gets no benefit from `Tearing`.  Trying to make it MSA-aware would probably let it solve the local instances on its own.  But the assignments introduced by `Tearing` can be simplified.

Reply:
I'm not following what you are saying. You say that `MemToReg` is not MSA-aware and therefore probably pessimistic in regards to MSA accesses, no? If so, then why is `Tearing` not enabling more optimizations?
What happens if you locally allocate an `bv64` but use `bv32` accesses to it? What will `MemToReg` do here?

Reply:
To detect MSAs `MemToReg` relies on `allocationType`, which `Tearing` cannot relyably update without copying the infrastructure of `MemToReg`.  I refered to the other approach, bringing a bit of `Tearing` into `MemToReg` instead.  This approach is now implemented.  It results in it promoting the local accesses on the first try, before `Tearing`.